src/copyful.tsx

```diff
@@ -1,18 +1,22 @@
-import React, { ReactElement } from 'react';
+import React, { useState, useEffect, ReactElement } from 'react';
 import { getInterpolatedCopy } from './helpers';
 
 type InterpolationContext = Record<string, string | number>;
 
 export const createCopyful = <TCopy extends object>(defaultCopy: TCopy) => {
-  const Context = React.createContext(defaultCopy);
-
-  const CopyfulProvider: React.FC = ({ children }) => (
-    <Context.Provider value={defaultCopy}>{children}</Context.Provider>
-  );
+  const Context = React.createContext<Partial<TCopy>>(defaultCopy);
+
+  const CopyfulProvider: React.FC<{ copy?: Partial<TCopy> }> = ({ children, copy }) => {
+    const [currentCopy, setCurrentCopy] = useState<Partial<TCopy>>(defaultCopy);
+    useEffect(() => {
+      setCurrentCopy(copy || defaultCopy);
+    }, [copy]);
```

Comment:
I don't think this is necessary because this component will re-render if the props change either way. This should be enough: 

```tsx
<Context.Provider value={copy || defaultCopy}>{children}</Context.Provider>;
```

Reply:
oh that's right. That's how we had it before it was removed.